AsyncDisplayKit/Layout/ASDimension.mm

```diff
@@ -17,32 +17,35 @@
 
 ASOVERLOADABLE ASDimension ASDimensionMake(NSString *dimension)
 {
-  // Handle empty string
-  if (dimension.length == 0) {
-    return ASDimensionMake(ASDimensionUnitPoints, 0.0);
-  }
-  
-  // Handle points
-  NSUInteger pointsStringLocation = [dimension rangeOfString:@"pt"].location;
-  if (pointsStringLocation != NSNotFound) {
-    // Check if points is at the end and remove it
-    if (pointsStringLocation == (dimension.length-2)) {
-      dimension = [dimension substringToIndex:(dimension.length-2)];
-      return ASDimensionMake(ASDimensionUnitPoints, dimension.floatValue);
+  if (dimension.length != 0) {
+    
+    // Handle points
+    NSUInteger pointsStringLocation = [dimension rangeOfString:@"pt"].location;
+    if (pointsStringLocation != NSNotFound) {
+      // Check if points is at the end and remove it
+      if (pointsStringLocation == (dimension.length-2)) {
+        dimension = [dimension substringToIndex:(dimension.length-2)];
+        return ASDimensionMake(ASDimensionUnitPoints, dimension.floatValue);
+      }
+    }
```

Comment:
Style: Use `hasSuffix:@"pt"]

Performance: remove `substringToIndex` – according to the docs, `floatValue` only checks if the string starts with a number.

Same below.